internals-js/src/operations.ts

```diff
@@ -1535,6 +1551,32 @@ export class SelectionSet {
     this._selections = mapValues(keyedSelections);
   }
 
+  minimizeSelectionSet(namedFragments: NamedFragments, fragmentSpreadsById: Map<string, FragmentSpreadSelection>): SelectionSet {
+    return this.lazyMap((selection) => {
+      if (selection.kind === 'FragmentSelection' && selection.element.typeCondition && selection.element.appliedDirectives.length === 0 && selection.selectionSet) {
+        const id = print(selection.toSelectionNode());
```

Comment:
I think `selection.toString()` should be the same?

Reply:
As discussed this also has a problem of ordering in the selection sets, i.e. given 
```graphql
... on Foo {
  foo
  bar
}
... on Foo {
  bar
  baz
}
```
they should be extracted as single fragment. Since this is printed selection set, this key can be pretty large and we should look into some better hashing approach.